xcode/Subconscious/Shared/Components/AppUpgradeView.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  AppUpgradeView.swift
+//  Subconscious
+//
+//  Created by Gordon Brander on 3/22/23.
+//
+
+import SwiftUI
+import Combine
+import ObservableStore
+
+/// Displays information to the user when app migration / rebuild happening.
+struct AppUpgradeView: View {
+    private let bgGradient = LinearGradient(
+        gradient: Gradient(
+            colors: [.brandBgTan, .brandBgGrey, .brandBgBlush]
+        ),
+        startPoint: .topLeading,
+        endPoint: .bottomTrailing
+    )
+    private let spinnerSize: CGFloat = 256
+    private let logoSize: CGFloat = 180
+    // Duration of certain completion transition animations
+    private let transitionDuration: CGFloat = 1
+
+    var state: AppUpgradeModel
+    var send: (AppUpgradeAction) -> Void
+
+    var body: some View {
+        VStack {
+            VStack {
+                if state.isComplete {
+                    Text("Welcome to Subconscious").transition(
+                        .asymmetric(
+                            insertion: .opacity.animation(
+                                .easeOut(duration: transitionDuration)
+                                .delay(transitionDuration)
+                            ),
+                            removal: .opacity.animation(.default)
+                        )
+                    )
+                } else {
+                    Text("What? Subconscious is evolving!").transition(
+                        .opacity.animation(
+                            .easeOut(duration: transitionDuration)
+                        )
+                    )
+                }
+                Spacer()
+            }
+            .foregroundColor(.brandBgBlack)
+            .multilineTextAlignment(.center)
+            .font(.title3)
+            .bold()
+            .frame(height: AppTheme.unit * 32)
+
+            VStack(alignment: .center) {
+                ProgressTorusView(
+                    isComplete: state.isComplete,
+                    size: spinnerSize
+                ) {
+                    Image("sub_logo_light")
+                        .resizable()
+                        .frame(width: logoSize, height: logoSize)
+                }
+                .shadow(color: .brandMarkPurple, radius: 72)
+                Spacer().frame(height: AppTheme.unit * 12)
+                if !state.isComplete {
+                    Text(verbatim: state.progressMessage)
+                        .id(state.progressMessage)
+                        .transition(.push(from: .bottom))
+                }
+            }
+            .italic()
+            .foregroundColor(.secondary)
+            .multilineTextAlignment(.center)
+            .frame(maxWidth: spinnerSize)
+
+            Spacer()
+
+            VStack(alignment: .center, spacing: AppTheme.unit) {
+                    Text("Upgrading your Subconscious.")
+                    Text("This might take a minute.")
+            }
```

Comment:
What happens if an error occurs during this process... Do we show anything to the user?